src/ctia/lib/riemann.clj

```diff
@@ -177,3 +169,17 @@
                                 :status "500"
                                 :error "true"})
             (throw e)))))))
+
+(defn start [context config]
+  (let [conn (-> (select-keys config
+                              [:host :port :interval-in-ms])
+                 riemann/tcp-client
+                 (riemann/batch-client
+                  (or (:batch-size config) 10)))
+        service-prefix (or (:service-prefix config) "CTIA")]
+    (assoc context
+           :conn conn
+           :service-prefix service-prefix)))
+
+(defn stop [{:keys [^RiemannClient conn]}]
+  (.close conn))
```

Comment:
I would check if conn is not nil. I think that the stop function can be invoked after the TK `init` and before the `start`

Reply:
good catch, thanks